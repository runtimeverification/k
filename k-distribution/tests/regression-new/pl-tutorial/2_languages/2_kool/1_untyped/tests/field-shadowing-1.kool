// Testing shadowing of a field by another field and
// whether the object stack environment works.

class C1 {
  var x, y;
  method C1() {}
  method setx1(v) { x = v; }
  method sety1(v) { y = v; }
  method getx1() { return x; }
  method gety1() { return y; }
}

class C2 extends C1 {
  var y;
  method C2() {}
  method sety2(v) { y = v; }
  method getx2() { return x; }
  method gety2() { return y; }
}

class Main {
  method Main() {
    var o2 = new C2();
    o2.setx1(11);
    o2.sety1(12); 
    o2.sety2(99);
    print(o2.getx1(), " ");
    print(o2.gety1(), " ");
    print(o2.getx2(), " ");
    print(o2.gety2(), "\n");
  }
}

// 11 12 11 99
