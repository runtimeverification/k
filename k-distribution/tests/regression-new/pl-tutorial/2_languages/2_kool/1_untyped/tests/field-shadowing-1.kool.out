11 12 11 99
